reference/forms/types/collection.rst

```diff
@@ -273,6 +273,15 @@ form you have to set this option to true. However, existing collection entries
 will only be deleted if you have the allow_delete_ option enabled. Otherwise
 the empty values will be kept.
 
+.. caution::
+
+    The ``delete_empty`` option only removes items when the normalized value is
+    ``null``. If your `type`_ is a compound form type, you must either set the
```

Comment:
Then we should be even more explicit it here, before 2.8 `options` may be confusing, let's use:
```rst
If the nested `type`_ is a compound form type
```
WDYT?